src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -60,6 +57,12 @@
     Text bracketCloseText;
     Text plusText;
 
+    private static final String APPEND_TO_WATCH_LIST_FORMAT = "%s;%s#%d";
+    private static final String CREATE_WATCH_LIST_FORMAT = "id:%s#%d";
```

Comment:
I was thinking of that, initially. But I could not figure out a better solution than using `instanceof` since `id:a#b` will be expanded to a conjunction of repo and id qualifier.

Another way would be to create another method in FilterExpression to check for it, but I feel that it's not what FilterExpression class is supposed to do :s